tests/framework/results/test_interface.py

```diff
@@ -173,7 +173,7 @@ def mock__prepare_population(self, event):
     return population
 
 
-def test_integration_full_observation(mocker):
+def test_interface_full_observation(mocker):
```

Comment:
It seems like this test would be better as a proper integration test (i.e. with an `InteractiveContext`).

Reply:
This test is a weird one. What it seems like it's really testing to me is whether or not aggregators get called on the event that they're registered to ("a_measure" is called because it's registered to  "collect_metrics" but "another_measure" is NOT called b/c it's registered to "time_step__prepare"). 

I think there's some utility to this, though it does seem undercooked. 

I'll try and beef it up as well as rename it I think b/c it's not really a "full_observation" test imo

Reply:
@rmudambi What do you think of this updated test? I changed the name to not be misleading. I'm happy to add a proper integration test but it sort of seems unnecessary since in the following PR I have near-e2e tests using `InteractiveContext` which checks the formatting of the output results.